pkg/cloud/aws/services/ec2/natgateways.go

```diff
@@ -165,21 +172,32 @@ func (s *Service) createNatGateway(subnetID string) (*ec2.NatGateway, error) {
 		return nil, errors.Wrapf(err, "failed to create IP address for NAT gateway for subnet ID %q", subnetID)
 	}
 
-	out, err := s.scope.EC2.CreateNatGateway(&ec2.CreateNatGatewayInput{
+	createInput := &ec2.CreateNatGatewayInput{
 		SubnetId:     aws.String(subnetID),
 		AllocationId: aws.String(ip),
-	})
-
-	if err != nil {
+	}
+	var out *ec2.CreateNatGatewayOutput
+	create := func() (bool, error) {
+		if out, err = s.scope.EC2.CreateNatGateway(createInput); err != nil {
+			return false, err
+		}
+		return true, nil
+	}
+	if err := wait.WaitForWithRetryable(wait.NewBackoff(), create, awserrors.InvalidSubnet); err != nil {
```

Comment:
Is this the only valid retryable error? What about the other ones related to invalid subnets (there are 2 or 3 more)? Where did you get your list from (I had trouble finding anything more than the generic/common list of errors)?

Reply:
For each of the errors, I tested against the AWS API to see what errors where returned, since the docs are spotty.

Reply:
Painful 😦